roapi-http/src/api/mod.rs

```diff
@@ -50,18 +50,12 @@ pub fn bytes_to_json_resp(bytes: Vec<u8>) -> Response<Body> {
     bytes_to_resp(bytes, "application/json")
 }
 
-pub fn encode_type_from_hdr(
-    headers: header::HeaderMap,
-) -> Result<encoding::ContentType, ApiErrResp> {
+pub fn encode_type_from_hdr(headers: header::HeaderMap) -> encoding::ContentType {
     match headers.get(header::ACCEPT) {
-        None => Ok(encoding::ContentType::Json),
-        Some(hdr_value) => {
-            encoding::ContentType::try_from(hdr_value.as_bytes()).map_err(|_| ApiErrResp {
-                code: http::StatusCode::BAD_REQUEST,
-                error: "unsupported_content_type".to_string(),
-                message: format!("{:?} is not a supported response content type", hdr_value),
-            })
-        }
+        None => encoding::ContentType::Json,
+        Some(hdr_value) => encoding::ContentType::try_from(hdr_value.as_bytes())
+            .or::<()>(Ok(ContentType::Json))
+            .unwrap(),
```

Comment:
How about simplify this with unwrap_or?

```suggestion
        Some(hdr_value) => encoding::ContentType::try_from(hdr_value.as_bytes()).unwrap_or(ContentType::Json),
```